x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/plugin/SqlClearCursorResponseListener.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.sql.plugin;
+
+import org.elasticsearch.rest.BytesRestResponse;
+import org.elasticsearch.rest.RestChannel;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.rest.RestResponse;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.rest.action.RestResponseListener;
+import org.elasticsearch.xcontent.MediaType;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentType;
+import org.elasticsearch.xpack.sql.action.SqlClearCursorRequest;
+import org.elasticsearch.xpack.sql.action.SqlClearCursorResponse;
+
+public class SqlClearCursorResponseListener extends RestResponseListener<SqlClearCursorResponse> {
+
+    RestRequest request;
+    private final MediaType mediaType;
+
+    protected SqlClearCursorResponseListener(RestChannel channel, RestRequest request, SqlClearCursorRequest sqlRequest) {
+        super(channel);
+        this.request = request;
+        this.mediaType = SqlMediaTypeParser.getResponseMediaType(request, sqlRequest);
```

Comment:
Maybe `getResponseMediaType` could return an `XContentType` instead of the `assert` below?

Reply:
Probably the best thing here is to also support text formats (see SqlResponseListener)

Reply:
I added support for the text option (same as SqlResponseListener). 
It's returning nothing though, just a RestStatus.OK. Not sure if there is something more meaningful we can return.

Reply:
An alternative would be to simply not support text formats here.
The precedent being the `/translate` endpoint, that (rightfully) has no logic for this and will simply return XContent formats, even if text is required (and maybe surprisingly, not complained about).
The reasoning for it being that this works for the successful case only. In case of failure, this will either return a XContent message, even if text is required (for instance, the cursor is invalid/damaged), or an empty message for failure (ex.: the cursor has already expired), since the `response.succeeded` isn't conveyed (I guess this could be text encoded (`succeeded|true/false`, `succeeded,true/false`, but seems an overkill IMO).

Reply:
Similar comment to @bpintea and use the initial approach of `RestToXContentListener` just passing the proper XContent (CBOR) if the binary communication is set.